File: HashMap/2428-equal-row-and-column-pairs/equal-row-and-column-pairs.java
class Solution {
    public int equalPairs(int[][] grid) {
        
        int n = grid.length;

        int count = 0;
        for (int i=0;i<n;i++) {
            int[] row = grid[i];
            for (int k=0;k<n;k++) {
                int[] col = new int[n];

                for(int j = 0; j<n;j++) {
                    col[j] = grid[j][k];
                }
                
                if(Arrays.equals(col, row)) {
                    count++;
                }
            }
            

           
        }
        return count;
    }
}

File: Array/2428-equal-row-and-column-pairs/equal-row-and-column-pairs.py
class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:
        row_count = {}
    
        count = 0
        n = len(grid)
        for i in range(n):
            t = tuple(grid[i][:])
            if t in row_count:
                row_count[t]+=1
            else:
                row_count[t]=1


        for i in range(n):
            t = ()
            for j in range(n):
                t += (grid[j][i], )
            print(t)
            if t in row_count:
                count+=row_count[t]
            
        return count



        
